File: ExtremeCore/Classes and Modules/ConfigsHandler.vb
Imports System.IO
Imports Newtonsoft.Json

Public Class ConfigsHandler
    Private _values As New Dictionary(Of String, Object)
    Private ReadOnly _autoSave As Boolean
    Private ReadOnly _dataFile As String

    Public Sub New(filePath As String, Optional defaultData As String = "", Optional isAutoSave As Boolean = False)
        _autoSave = isAutoSave
        _dataFile = filePath

        If File.Exists(_dataFile) = False And defaultData <> "" Then
            My.Computer.FileSystem.WriteAllText(_dataFile, defaultData, False)
        End If

        If File.Exists(_dataFile) = True Then
            Load()
        End If
    End Sub

    Public Default Property Item(key As String) As Object
        Get
            Return _values(key)
        End Get
        Set
            _values(key) = Value

            If _autoSave Then
                Save()
            End If
        End Set
    End Property

    Public Sub Save()
        Dim json As String = JsonConvert.SerializeObject(_values, Formatting.Indented)
        File.WriteAllText(_dataFile, json)
    End Sub

    Public Sub Delete(key As String)
        _values.Remove(key)

        If _autoSave Then
            Save()
        End If
    End Sub

    Private Sub Load()
        Dim json As String = File.ReadAllText(_dataFile)
        _values = JsonConvert.DeserializeObject(Of Dictionary(Of String, Object))(json)
    End Sub
End Class

File: ExtremeCore/Classes/ConfigsHandler.cs
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

namespace ExtremeCore.Classes
{
    public class ConfigsHandler
    {
        private Dictionary<string, object> _values = new Dictionary<string, object>();
        private readonly bool _autoSave;
        private readonly string _dataFile;

        public ConfigsHandler(string filePath, string defaultData = "", bool isAutoSave = false)
        {
            _autoSave = isAutoSave;
            _dataFile = filePath;

            if (File.Exists(_dataFile) == false & defaultData != "")
                File.WriteAllText(_dataFile, defaultData);

            if (File.Exists(_dataFile))
                Load();
        }

        public object this[string key]
        {
            get => _values[key];
            set
            {
                _values[key] = value;

                if (_autoSave)
                    Save();
            }
        }

        public void Save()
        {
            string json = JsonConvert.SerializeObject(_values, Formatting.Indented);
            File.WriteAllText(_dataFile, json);
        }

        public void Delete(string key)
        {
            _values.Remove(key);

            if (_autoSave)
                Save();
        }

        private void Load()
        {
            string json = File.ReadAllText(_dataFile);
            _values = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
        }
    }
}
